### PR TITLE
Setup dev-build-tools package and add Rollup Translations plugin with tests

### PR DESCRIPTION
**Pull request by filipsobol**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature (build-tools): Create a new `@ckeditor/ckeditor5-dev-build-tools` package containing a plugin for preparing CKEditor 5 translations using Rollup. See ckeditor/ckeditor5#15712.

Closes ckeditor/ckeditor5#15712.

---

> [!CAUTION]  
> This PR targets `epic-kowalski` branch.

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

### CODE REVIEW

File: packages/ckeditor5-dev-build-tools/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/**
```
Comment (filipsobol) [inline packages/ckeditor5-dev-build-tools/rollup.config.js:1]: This Rollup configuration file is used to build this package itself.

File: packages/ckeditor5-dev-build-tools/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// TODO: When it's close to being production ready, remove the comment below.
```
Comment (filipsobol) [inline packages/ckeditor5-dev-build-tools/src/config.ts:1]: For now, this Rollup configuration file is only added for testing purposes. Review of it is optional.

**Review by pszczesniak — COMMENTED** [review]
Nothing obvious caught my eye.

**Comment by pomek** [comment]
As we talked via internal channels, there is only one question: will we create more packages? Depending on the answer, there are a few scenarios we need to consider:

* If so (more packages): let's keep everything here and adjust processes (CI, release, development, etc.).
* Otherwise (just a single package): could we keep it outside the cke5-dev repo?

Let me explain the second option. The `ckeditor5-dev` repository and all packages inside are not ready for TypeScript. I want to avoid mixing JS and TS code within a single repository. I would like to rewrite packages from this repo to TypeScript soon and make some clean-up. It would also require aligning processes (mentioned above) to enable easy development of new features and make everything compatible with the main project that consumes almost all of those tools.

File: package.json [hunk 3]
```diff
@@ -43,10 +45,6 @@
       "eslint --quiet"
     ]
   },
-  "eslintIgnore": [
-    "coverage/**",
-    "packages/*/node_modules/**"
-  ],
```
Comment (filipsobol) [inline package.json:49]: Moved to `.eslintrc.js`

**Comment by filipsobol** [comment]
PR is ready for review. I've updated the test command, linter configuration, and the release process to work with TypeScript and its build step.

**Comment by filipsobol** [comment]
This PR targets the `epic-kowalski` branch, so we can continue to add features and test them as we go, without affecting the release process of the rest of the packages.

When we are ready and feel that the new package is stable, we will merge the `epic-kowalski` branch into `master`.

File: packages/ckeditor5-dev-build-tools/package.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "@ckeditor/ckeditor5-dev-build-tools",
+  "version": "0.1.0",
```
Comment (pomek) [inline packages/ckeditor5-dev-build-tools/package.json:3]:
```suggestion
  "version": "0.0.1",
```

File: packages/ckeditor5-dev-build-tools/package.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  "description": "Rollup-based tools used to build CKEditor packages",
```
Comment (pomek) [inline packages/ckeditor5-dev-build-tools/package.json:4]:
```suggestion
  "description": "Rollup-based tools used to build CKEditor 5 packages.",
```

**Pull request merged**